arangod/RestServer/UpgradeFeature.cpp

```diff
@@ -285,9 +285,16 @@ void UpgradeFeature::upgradeLocalDatabase() {
   bool ignoreDatafileErrors = databaseFeature.ignoreDatafileErrors();
 
   for (auto& name : databaseFeature.getDatabaseNames()) {
-    TRI_vocbase_t* vocbase = databaseFeature.lookupDatabase(name);
+    auto vocbase = databaseFeature.useDatabase(name);
+
+    // in this phase, all databases returned by getDatabaseNames() should
```

Comment:
Is this about UpgradeFeature?
If not that's not true. getDatabaseNames return some snapshot, which can be outdated

Reply:
There should not be any concurrent deletions while this code is executed. The UpgradeFeature only runs at server startup.
The previous version of the code called `lookupDatabase(...)` and then unconditionally dereferenced the return value, which was unsafe. The version here avoids any potential issues by checking the return value before deferencing them.